lib/Runtime/Provisioner/Provisioner.cpp

```diff
@@ -101,11 +101,14 @@ llvm::Error Provisioner::provision(DAGListTy &networks, Module &module,
       // compiled the function reuse it.
       auto it = functions_.find(node->name);
       if (it == functions_.end()) {
+        // Copy BackendOptions and add the compiler hints for this function.
+        auto options = cctx.backendOpts;
```

Comment:
Why do we need a copy here? Is it to prevent the backendHints from leaking back out to the caller?